common/src/chain/pow.rs

```diff
@@ -0,0 +1,112 @@
+use crate::chain::config::ChainType;
+use crate::chain::ChainConfig;
+use crate::Uint256;
+use std::time::Duration;
+
+/// Chain Parameters for Proof of Work, as found in
+/// https://github.com/bitcoin/bitcoin/blob/eca694a4e78d54ce4e29b388b3e81b06e55c2293/src/chainparams.cpp
+pub struct PoWChainConfig {
+    no_retargeting: bool,
+    allow_min_difficulty_blocks: bool,
+    limit: Uint256,
+}
+
+impl ChainConfig {
+    pub fn get_pow_config(&self) -> PoWChainConfig {
+        PoWChainConfig {
+            no_retargeting: no_retargeting(&self.chain_type()),
+            allow_min_difficulty_blocks: allow_min_difficulty_blocks(&self.chain_type()),
+            limit: limit(&self.chain_type()),
+        }
+    }
+}
+
+impl PoWChainConfig {
+    pub fn no_retargeting(&self) -> bool {
+        self.no_retargeting
+    }
+
+    pub fn allow_min_difficulty_blocks(&self) -> bool {
+        self.allow_min_difficulty_blocks
+    }
+
+    pub fn limit(&self) -> Uint256 {
+        self.limit
+    }
+
+    pub fn target_timespan(&self) -> Duration {
+        Duration::new(14 * 24 * 60 * 60, 0)
+    }
+
+    pub fn target_spacing(&self) -> Duration {
+        Duration::new(10 * 60, 0)
+    }
+
+    /// https://github.com/bitcoin/bitcoin/blob/eca694a4e78d54ce4e29b388b3e81b06e55c2293/src/pow.cpp#L56
```

Comment:
I'm going to be a nitpicker again. This is a doc comment and as such will show up in documentation when you generate it using `cargo doc`. The first line is the main brief description of what the function does and appears in lists of functions and such. It is quite ugly to have just a URL there.